### PR TITLE
Verify issues resolved before 2.39

### PR DESCRIPTION
**Pull request by abayer**
Replaces #160, #161, #162 

Rollup of test verifications for a number of issues:
* [JENKINS-38052](https://issues.jenkins-ci.org/browse/JENKINS-38052)
* [JENKINS-27916](https://issues.jenkins-ci.org/browse/JENKINS-27916) (sorta)
* [JENKINS-41248](https://issues.jenkins-ci.org/browse/JENKINS-38052)
* [JENKINS-28321](https://issues.jenkins-ci.org/browse/JENKINS-28321)

cc @reviewbybees

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java [hunk 1]
```diff
@@ -224,4 +224,68 @@ public void traitsSandbox() throws Exception {
...
+    @Issue("JENKINS-27916")
+    @Test
+    public void gStringInMapKey() throws Exception {
+        WorkflowJob job = jenkins.jenkins.createProject(WorkflowJob.class, "p");
+        job.setDefinition(new CpsFlowDefinition("def s1 = \"first-${env.BUILD_NUMBER}\"\n" +
+                "def s2 = \"second-${env.BUILD_NUMBER}\"\n" +
+                "def m = [(s1): 'first-key',\n" +
+                "  \"${s2}\": 'second-key',\n" +
+                "  \"third-${env.BUILD_NUMBER}\": 'third-key']\n" +
+                "m.each { k, v -> echo \"${k}:${v}\" }\n", true));
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java]: As noted elsewhere, this is _not_ what the issue is about.

File: src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java [hunk 2]
```diff
@@ -224,4 +224,68 @@ public void traitsSandbox() throws Exception {
...
+                "    void setA(int a) {\n" +
+                "        this.a = a\n" +
+                "    }\n" +
+                "    String getA() {\n" +
+                "        return a\n" +
+                "    }\n" +
+                "}\n" +
+                "Foo foo = new Foo()\n" +
+                "foo.setA(10)\n" +
+                "echo \"a is ${foo.getA()}\"", true));
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java]: Covered by https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/29?

File: src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java [hunk 3]
```diff
@@ -224,4 +224,68 @@ public void traitsSandbox() throws Exception {
...
+    @Test
+    public void whitelistedMethodPointer() throws Exception {
+        WorkflowJob job = jenkins.createProject(WorkflowJob.class, "p");
+        job.setDefinition(new CpsFlowDefinition("def foo = 'original'\n" +
+                "def bar = foo.&toUpperCase\n" +
+                "echo bar.call()\n", true));
+
+        WorkflowRun b = jenkins.buildAndAssertSuccess(job);
+
+        jenkins.assertLogContains("ORIGINAL", b);
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java]: Will fail if run in Turkey or Azerbaijan: `ORİGİNAL`

Comment (abayer) [inline src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java]: I...huh.

**Comment by svanoort** [comment]
AFAICT looks fine aside from the issues Jesse noted

**Pull request merged**